receiver/prometheusreceiver/internal/logger.go

```diff
@@ -120,20 +117,20 @@ func matchLogLevel(key interface{}, val interface{}) (level.Value, bool) {
 }
 
 // find a matching zap logging function to be used for a given level
-func levelToFunc(logger *zap.SugaredLogger, lvl level.Value) (func(string, ...interface{}), error) {
+func levelToFunc(logger *zap.SugaredLogger, lvl level.Value) func(string, ...interface{}) {
```

Comment:
Q.: Does the tool detects all paths returning nil?

Reply:
yes